src/cachingreader.cpp

```diff
@@ -301,7 +301,7 @@ void CachingReader::process() {
     }
 }
 
-int CachingReader::read(int sample, int num_samples, CSAMPLE* buffer) {
+int CachingReader::read(int sample, bool reverse, int num_samples, CSAMPLE* buffer) {
```

Comment:
This change allows the RAMAN to request the samples in the desired order and saves it from an extra loop though the buffer to in place revert it.